cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+
+    @property
+    def is_displayed(self):
+        return self.in_explorer
+
+
+class DashboardCards(View):
+    not_started_plans = MigrationDashboardStatusCard("Not Started")
+    in_progress_plans = MigrationDashboardStatusCard("In Progress")
+    completed_plans = MigrationDashboardStatusCard("Complete")
+    archived_plans = MigrationDashboardStatusCard("Archived")
+
+    def read(self):
+        return [c.read().get("name") for c in self.sub_widgets if c.is_active]
+
+
+class MigrationPlanView(MigrationView):
+    dashboard_cards = View.nested(DashboardCards)
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    title = Text('.//div[contains(@class, "pull-left")]//h3')
+    create_migration_plan = Text(locator='(//a|//button)[text()="Create Migration Plan"]')
+    configure_providers = Text(locator='//a[text()="Configure Providers"]')
+    sort_type_dropdown = SelectorDropdown("id", "sortTypeMenu")
+    sort_direction = Text(locator=".//span[contains(@class,'sort-direction')]")
+    progress_card = MigrationProgressBar()
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+
+    @property
+    def in_migration_plan(self):
+        return (
+            self.in_explorer and self.create_migration_plan.is_displayed and
+            len(self.browser.elements('.//div[contains(@class,"card-pf")]')) > 0
+            and len(self.browser.elements(PFIcon.icons.WARNING)) < 1
+        )
+
+    @property
+    def is_displayed(self):
+        return self.in_migration_plan
+
+
+class NotStartedPlansView(MigrationPlanView):
+    plans_not_started_list = MigrationPlansList("plans-not-started-list")
+
+    @property
+    def is_displayed(self):
+        return (self.in_migration_plan and
+                self.plans_not_started_list.is_displayed and
+                self.title.text == 'Not Started Plans')
+
+
+class InProgressPlansView(MigrationPlanView):
+    progress_card = MigrationProgressBar()
+
+    @property
+    def is_displayed(self):
+            return (self.in_migration_plan and
+                    self.progress_card.is_displayed and
+                    self.title.text == 'In Progress Plans')
+
+
+class CompletedPlansView(MigrationPlanView):
+    plans_completed_list = MigrationPlansList("plans-complete-list")
+
+    @property
+    def is_displayed(self):
+            return (self.in_migration_plan and
+                    self.plans_completed_list.is_displayed and
+                    self.title.text == 'Completed Plans')
+
+
+class ArchivedPlansView(MigrationPlanView):
+    archived_plans_list = MigrationPlansList("plans-archived-list")
+
+    @property
+    def is_displayed(self):
+            return (self.in_migration_plan and
+                    self.archived_plans_list.is_displayed and
+                    self.title.text == 'Archived Plans')
+
+
+class AddMigrationPlanView(View):
```

Comment:
**Optional**

The definition of this class is nearly 200 lines, consider defining some of the nested view classes outside of `AddMigrationPlanView`, and then nest them with `View.nested()` calls.

Reply:
It's smaller now . Removed unused code